Source/monster.cpp

```diff
@@ -1128,6 +1128,8 @@ void StartMonsterDeath(int i, int pnum, bool sendmsg)
 	M_FallenFear(monster.position.tile);
 	if ((monster.MType->mtype >= MT_NACID && monster.MType->mtype <= MT_XACID) || monster.MType->mtype == MT_SPIDLORD)
 		AddMissile(monster.position.tile, { 0, 0 }, Direction::South, MIS_ACIDPUD, TARGET_PLAYERS, i, monster._mint + 1, 0);
+	if (monster._mmode == MonsterMode::Petrified)
+		monster.Petrify();

```

Comment:
This won't work as line 1119 always sets the `_mmode = Death`, but instead you could make that line conditional on `_mmode != Petrified`

Reply:
Very good find. Thank you!

Reply:
You can remove this now that the mode isn't changed :)